src/CoreBundle/Bridge/Chameleon/TableEditor/DataMailProfileTableEditor.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace ChameleonSystem\CoreBundle\Bridge\Chameleon\TableEditor;
+
+
+use ChameleonSystem\CoreBundle\Interfaces\FlashMessageServiceInterface;
+use ChameleonSystem\CoreBundle\ServiceLocator;
+use Psr\Log\LoggerInterface;
+use TCMSTableEditor;
+use Twig\Error\SyntaxError;
+use Twig\Environment;
+use Twig\Source;
+
+class DataMailProfileTableEditor extends TCMSTableEditor
+{
+    const MESSAGE_MANAGER_CONSUMER = 'DataMailProfileTableEditorMessages';
+
+    protected function DataIsValid(&$postData, $oFields = null)
+    {
+        $twig = $this->getTwigEnvironment();
+        try {
+            $twig->tokenize(new Source($postData['body'], 'body'));
+            $twig->tokenize(new Source($postData['body_text'], 'body_text'));
+        }catch (SyntaxError $e) {
+
+            $this->getLogger()->error(
+                sprintf('failed to pares body field in E-Mail Template %s', $postData['name']),
+                [
+                    'sFieldName' => 'body',
+                    'message' => $e->getMessage(),
+                    'guess' => $e->guess(),
+                    'context' => $e->getSourceContext()->getCode(),
+                    'exception' => $e,
+                ]
+            );
+
+            $this->getFlashMessageService()->addMessage(
+                self::MESSAGE_MANAGER_CONSUMER,
+                'chameleon_system.table_editor_twig_error',
+                [
+                    'sFieldName' => 'body',
+                    'message' => $e->getMessage(),
+                    'guess' => $e->guess(),
+                ]
+            );
+            return false;
+        }
```

Comment:
This will always mark the `body` field with an error, even if an error occurs in the the `body_text` field. This can be very misleading. We should make sure that the error always points to the exact field where the error occured.

Reply:
correct, fixed.